nilearn/input_data/tests/test_nifti_masker.py

```diff
@@ -110,6 +111,37 @@ def test_mask_3d():
         assert_raises(TypeError, masker.fit)
 
 
+def test_mask_4d():
+    # Dummy mask
+    mask = np.zeros((10, 10, 10))
+    mask[3:7, 3:7, 3:7] = 1
+    mask_img = Nifti1Image(mask, np.eye(4))
+    n_mask_vox = len(np.where(mask == True)[0])
+
+    # Dummy data
+    data = np.zeros((10, 10, 10, 3))
+    data[..., 0] = 1
+    data[..., -1] = 1
+    data_img_4d = Nifti1Image(data, np.eye(4))
+    data_imgs = [Nifti1Image(data[..., 0], np.eye(4)),
+        Nifti1Image(data[..., 1], np.eye(4)),
+        Nifti1Image(data[..., 2], np.eye(4))]
+
+    # check whether transform is indeed selecting niimgs subset
+    mask_time = np.array([0, 2])
+    masker5 = NiftiMasker(mask_img=mask_img, mask_time=mask_time)
+    masker5.fit()
+    data_trans = masker5.transform(data_imgs)
+    assert_true(np.all(data_trans))
+    assert_true((2, n_mask_vox) == data_trans.shape)  # verify mask application
+
+    masker5 = NiftiMasker(mask_img=mask_img, mask_time=mask_time)
+    masker5.fit()
+    data_trans = masker5.transform(data_img_4d)
+    assert_true(np.all(data_trans))
+    assert_true((2, n_mask_vox) == data_trans.shape)  # verify mask application
```

Comment:
Could we not check that applying index_img and then the mask without time_mask is the same as applying the mask with time_mask to the original image?